nomos-services/data-availability/src/network/mod.rs

```diff
@@ -0,0 +1,25 @@
+// std
+// crates
+use futures::Stream;
+use tokio::sync::oneshot;
+// internal
+use nomos_network::backends::NetworkBackend;
+use nomos_network::NetworkService;
+use overwatch_rs::services::relay::OutboundRelay;
+use overwatch_rs::services::ServiceData;
+
+#[async_trait::async_trait]
+pub trait NetworkAdapter {
+    type Backend: NetworkBackend + 'static;
+
+    type Blob: Send + Sync + 'static;
+    type Reply: Send + Sync + 'static;
+
+    async fn new(
+        network_relay: OutboundRelay<<NetworkService<Self::Backend> as ServiceData>::Message>,
+    ) -> Self;
+
+    async fn blob_stream(
+        &self,
+    ) -> Box<dyn Stream<Item = (Self::Blob, oneshot::Sender<Self::Reply>)> + Unpin + Send>;
```

Comment:
mmm, thinking how the reply would work
Is your idea that the adapter somehow knows this and can create that channel on its own? 
In that case, I think it should also be responsible for sending the blob, or better, we should unify the responsibilities of sending and interpreting the same message

Reply:
? Is your idea that the adapter somehow knows this and can create that channel on its own?
yes, this was just the first idea. It is probably gonna change. As for the first version we would be broadcasting the replies anyway.

I think actually for now Im just gonna remove this for what I explained above. And add a call to send the reply.

Reply:
> Is your idea that the adapter somehow knows this and can create that channel on its own?

Yes, or at least there's a single component that knows how to read and write those. 
What I want to avoid is that we define reading in one component and writing in another one, because we've know created hidden coupling between them

Reply:
Absolutely right. And the network adapter should not know about it.